Trees/BinarySearchTree.cpp

```diff
@@ -0,0 +1,358 @@
+/*
+    Author:
+    Varun Jayathirtha (www.github.com/varunjm)
+
+    Binary Search Tree:
+    A graph data structure where each node has a unidirectional path to at most
+    two node, none of which are itself or its predecessors, i.e the graph is
+    acyclic. The first node is called the root node. Another important feature
+    of this tree is that each node has values lesser than or equal to itself in
+    the left subtree and values greater than or equal to itself in the right
+    subtree.
+*/
+
+#include "BinarySearchTree.hpp"
+#include <iostream>
+#include <climits>
+#include <stack>
+#define INF INT_MAX
+
```

Comment:
By using a `const` you would be doing the same thing that you are doing with `#define` - giving the constant a better name, but in a more standard way for C++ (`#define` is old C style, modern C has `const` for the same purpose).